src/caffe/proto/caffe.proto

```diff
@@ -1231,4 +1231,11 @@ message PReLUParameter {
   optional FillerParameter filler = 1;
   // Whether or not slope paramters are shared across channels.
   optional bool channel_shared = 2 [default = false];
+  // Whether or not to constrain the negative slope between min_neg_slope and
+  // max_neg_slope.
+  optional bool constrain_neg_slope = 3 [default = true];
```

Comment:
Right. But it has no effect once a model is trained. If a trained model is just deployed, its behavior  won't change. So to the extent that it changes behavior, it won't be until after they've done some more training. At which point they would be evaluating before deploying it anyway. 

I don't mind changing the default to false. But I doubt most people will be aware that the option is there, and so will continue suffering along with crazy Prelu values (I was getting NaNs, and after instrumenting found it was due to an inf coming out of a Prelu in forward, going in to a softmax that generated NaN when dividing inf by inf. So if you're worried about hard-to-debug problems, I think a wacky negative slope is a bigger threat than a Prelu that's constrained to be somewhere in-between a ReLU (neg-slope == 0.0) and a linear unit (neg slope == 1.0).